pulsar-client-cpp/tests/PulsarFriend.h

```diff
@@ -57,6 +57,19 @@ class PulsarFriend {
         return std::static_pointer_cast<ConsumerStatsImpl>(consumerImpl->consumerStatsBasePtr_);
     }
 
+    // just param is MultiTopicConsumerImpl
```

Comment:
This comment is confused. I think you want to mean it should only be called when the underlying consumer of `consumer` is `MultiTopicConsumerImpl`?

If yes, please remove the confused comment and use `dynamic_cast` and null check instead of `static_cast` in the implementation. `dynamic_cast` can check the RTTI info so that it will return null if the cast is invalid. e.g.

```c++
#include <iostream>
using namespace std;

struct Base {
    virtual void f() {}
};

struct Derived : Base {
    void f() override {}
};

void test(Base* base) {
    cout << "dynamic_cast: " << dynamic_cast<Derived*>(base)
         << "\nstatic_cast: " << static_cast<Derived*>(base) << endl;
    delete base;
}

int main(int argc, char* argv[]) {
    test(new Base);
    test(new Derived);
    return 0;
}
```

You can see the difference from the output:

```
dynamic_cast: 0x0
static_cast: 0x600001eac020
dynamic_cast: 0x600001eac020
static_cast: 0x600001eac020
```